src/state/action-creators.js

```diff
@@ -434,26 +439,28 @@ export const lift = (id: DraggableId,
   windowScroll: Position,
   isScrollAllowed: boolean,
 ) => (dispatch: Dispatch, getState: Function) => {
-  (() => {
-    const state: State = getState();
-    // quickly finish any current animations
-    if (state.phase === 'DROP_ANIMATING') {
-      if (!state.drop || !state.drop.pending) {
-        console.error('cannot flush drop animation if there is no pending');
-        dispatch(clean());
-        return;
-      }
-      dispatch(completeDrop(state.drop.pending.result));
+  // Quickly finish any current drop animations
+  const initial: State = getState();
+
+  if (initial.phase === 'DROP_ANIMATING') {
+    if (!initial.drop || !initial.drop.pending) {
+      console.error('cannot flush drop animation if there is no pending');
+      dispatch(clean());
+    } else {
+      dispatch(completeDrop(initial.drop.pending.result));
     }
-  })();
+  }
 
   // https://github.com/chenglou/react-motion/issues/437
   // need to allow a flush of react-motion
+  dispatch(prepare());
+
   setTimeout(() => {
     const state: State = getState();
 
-    if (state.phase !== 'IDLE' && state.phase !== 'DRAG_COMPLETE') {
-      dispatch(clean());
+    // drag cancelled before timeout finished
+    if (state.phase !== 'PREPARING') {
```

Comment:
Previously we had no way of knowing if the drag was ended before this first timeout was fired. It is possible to do with a well orchestrated force press in touch dragging. By adding an extra phase we can know if something has changed in the elapsed time.